examples/basic_infection/README.md

```diff
@@ -0,0 +1,223 @@
+```
+init(context) {
+    context.observe_person_property_event(handle_person_property_change(context, person_id, previous_infection_status));
+}
+```
+
+The method `handle_person_property_change` writes a new line to the report file with the change in the person property of infection status.
+```
+fn handle_person_property_change(context, person_id, infection_status){
+    report_data = (t = context.get_time(), person_property_type = Type(infection_status), person_property_value = context.get_infection_status(person_id), person_id = person_id);
+    context.print_report_data(report_data); // Method to print to csv, tsv to be implemented in the
+}
+```
+
+### Periodic report for current value in person properties (Infection Status)
+To report the current state of each Infection Status (Susceptible, Infected, or Recovered), this report needs an additional data structure that keeps a count of individuals on each state and is updated every time an event is released due to changes in infection status.
+```
+person_property_counter = struct {
+    map: hashmap<PropertyType, hashmap<property_value, int>>;
+    // e.g., this map could represent 100 susceptible people
+    // <InfectionStatus, <Susceptible, 100>>
+}
+context.add_data_container(person_property_counter)
+```
+At initialization, the property counter is initialized with the current state of all the population. The report also observes changes in person properties which are handled by a function that updates counts in the property counter. The first report day is scheduled for t = 0.
+```
+init(context){
+    population = parameters.get_parameter(population);
+    for i in 0..population {
+        for person_property in context.get_person_property_types().to_list() {
+            for person_property_value in (person_property.to_list()) {
+                person_property_counter.increment(person_property,person_property_value);
+            }
+        }
+    }
+    context.observe_person_property_event(update_property_counter);
+    context.add_plan(report_periodic_item, time = 0);
+}
+```
+Methods are implemented for person property counter to increment and decrement the counters. For the periodic report, one method will release the periodic items to a report based on the period defined in parameters. Changes in the person properties are observed and the callback function `update_property_counter` updates the counts for each property.
+```
+fn update_property_counter(context, person_id,
+                           person_property_type,
+                           previous_person_property_value) {
+    person_property_counter.increment(context.get_person_property(person_property_type, person_id));
```

Comment:
I'm not specifying here how to implement the methods: increment, decrement, or `get_counter` for the `person_property_counter` because I think the important thing is that we want a counter that increments values for each person property type/value, but I can elaborate if necessary.